src/sql/workbench/api/common/extHostModelViewDialog.ts

```diff
@@ -716,11 +788,24 @@ export class ExtHostModelViewDialog implements ExtHostModelViewDialogShape {
 		this._onClickCallbacks.set(handle, callback);
 	}
 
-	public createDialog(title: string, dialogName?: string, extension?: IExtensionDescription, width?: azdata.window.DialogWidth): azdata.window.Dialog {
+	public createDialog(title: string, dialogName?: string, extension?: IExtensionDescription, width?: DialogWidth, dialogStyle?: DialogStyle, dialogPosition?: DialogPosition, suppressHeader?: boolean, suppressFooter?: boolean, dialogXYOffset?: IDialogXYOffset): azdata.window.Dialog {
+
 		let dialog = new DialogImpl(this, this._extHostModelView, this._extHostTaskManagement, extension);
+
 		if (dialogName) {
 			dialog.dialogName = dialogName;
 		}
+		if (dialogStyle) {
+			dialog.dialogStyle = dialogStyle;
+		}
+		if (dialogPosition) {
+			dialog.dialogPosition = dialogPosition;
+		}
+		if (dialogXYOffset) {
+			dialog.dialogXYOffset = dialogXYOffset;
+		}
+		dialog.suppressHeader = suppressHeader;
```

Comment:
Why are these not checked for validity?

In general though for optional args I would think that it would make more sense to have the dialog able to take in and handle undefined values and so these kind of checks would be unnecessary (it'd just default to some reasonable value if they're undefined)